sentry/src/db/campaign.rs

```diff
@@ -54,6 +59,114 @@ pub async fn fetch_campaign(
     Ok(row.as_ref().map(Campaign::from))
 }
 
+// TODO: Export this as it's the same as the one in channel list
+struct TotalCount(pub u64);
+impl<'a> FromSql<'a> for TotalCount {
+    fn from_sql(
+        ty: &Type,
+        raw: &'a [u8],
+    ) -> Result<Self, Box<dyn std::error::Error + Sync + Send>> {
+        let str_slice = <&str as FromSql>::from_sql(ty, raw)?;
+
+        Ok(Self(u64::from_str(str_slice)?))
+    }
+
+    // Use a varchar or text, since otherwise `int8` fails deserialization
+    accepts!(VARCHAR, TEXT);
+}
+
+pub async fn list_campaigns(
+    pool: &DbPool,
+    skip: u64,
+    limit: u32,
+    creator: &Option<Address>,
+    validator: &Option<ValidatorId>,
+    is_leader: &Option<bool>,
```

Comment:
- [ ] For types that implement `Copy` you can directly pass an owned `Option` (for Address, ValidatorId & bool)